### ISSUE TITLE
`__str__ method not found` Error with `ecr_embed` on `String::Builder.build` on 0.11.0

### ISSUE THREAD

**Issue by f** [manastech/crystal#2057]
```
String::Builder.build do |__str__|
    embed_ecr("hello.ecr", "__str__")
end
```
This code is broken on 0.11.0.

```
instantiating 'String::Builder:Class#build()'
in macro 'macro_4616040192' expanded macro: embed_ecr:1, line 1:

  1. __str__ << "Hello "
  2. (#<loc:"hello.ecr",1,10> env.params["name"] ).to_s __str__
  3. __str__ << "\n"

__str__ << "Hello "
^~~~~~~

undefined method '__str__' (If you declared '__str__' in a suffix if, declare it in a regular if for this to work. If the variable was declared in a macro it's not visible outside it)
```

**Comment by f** [comment]
**Update**

I solved with removing block.
```
__str__ = String::Builder.new
embed_ecr({{filename}}, "__str__")
__str__.to_s
```

And now I have another strange problem:

hello.ecr:
```
Hello <%= env.params["name"] %>
```

This is not working (says `env is not unknown`): 
```
kemal.add_route "GET", "/view/:name" do |env|
  render "hello.ecr"
end
```

This is working:
```
kemal.add_route "GET", "/view/:name" do |temp_env|
  env = temp_env
  render "hello.ecr"
end
```

**Comment by ysbaddaden** [comment]
What about `String.build` instead of `String::Builder.build` ?

**Comment by asterite** [comment]
@f Thanks for reporting this!

This and #2050 are probably worth a minor release with these fixes (0.11.1). All of these regressions are because of these two bug fixes: 81a7af2566187cb461ff5dfdc78ac26a67953751 and 92aabbd15a3928e774a463650d0e43e97f61fe02. There was a big chance that they introduced new bugs.

**Comment by f** [comment]
@ysbaddaden Same.
